src/SDKs/Batch/DataPlane/Azure.Batch.Unit.Tests/BatchSharedKeyCredentialUnitTests.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace Azure.Batch.Unit.Tests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Net.Http;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Batch.Protocol;
+    using Microsoft.Net.Http.Server;
+    using Xunit;
+
+    public class BatchSharedKeyCredentialUnitTests
+    {
+        private const string url = "http://localhost:2055";
+
+        [Theory]
+        [MemberData(nameof(HttpMethods))]
+        public async Task TestSharedKeySigning_ContentLength(HttpMethod httpMethod)
```

Comment:
The issue with a line in the sand name like "DoesNot" or "Does" with respect to including content length is that the behavior varies across target framework, so unless I conditionally compile the whole test out, it's going to be wrong for one of the frameworks when it comes to `DELETE`

I could add a bunch of tests clearly spelling out each case (either does or does not), but they would all just call into a common helper function (which is the whole point of `Theory`).

I've changed this to be: `HttpClient_IncludesContentLengthHeaderOnExpectedHttpVerbs`. Obviously what is "Expected" is still nebulous (you have to read the test code)...

Let me know if you don't think this is sufficient.